backend/src/test/java/de/benjaminranft/spotown/controller/DiscoveryControllerIntegrationTest.java

```diff
@@ -0,0 +1,169 @@
+                        new ArrayList<>(List.of("drink", "nature")))
+
+        ));
+
+        assertThat(response.getStatusCode(), is(HttpStatus.OK));
+        assertThat(response.getBody(), is(expectedList.toArray()));
+    }
+
+    @Test
+    @DisplayName("The add method should add a Discovery to User and return the added Discovery Object")
+    public void add(){
+
+        //GIVEN
+        String url = backendAccessLink();
+
+        AddDiscoveryDto addDiscoveryDto = new AddDiscoveryDto(
+                "Fun Discovery", "Fun address Str. 12, 20253 Hamburg", "https://google.com", "345678976543", "Fun notes fun notes fun notes fun notes", new ArrayList<>(
+                        List.of("restaurant","art"))
+        );
+
+        //WHEN
+        when(idUtils.generateId()).thenReturn("1234");
+        when(timestampUtils.generateTimestampEpochSeconds()).thenReturn(Instant.parse("2020-11-18T18:35:24.00Z"));
+
+        HttpEntity<AddDiscoveryDto> entity = getValidAuthorizationEntity(addDiscoveryDto);
+        ResponseEntity<Discovery> response = testRestTemplate.exchange(url, HttpMethod.POST, entity, Discovery.class);
+
+        Instant expectedTimestamp = Instant.parse("2020-11-18T18:35:24.00Z");
+        Object updatedUser = userDao.findById("heinz");
+        Object expectedUser = Optional.of(new User(
+                "heinz",
+                "$2a$10$0HZGmicEH786L.HeSIjhOuvIK3ixlYij4luVHBNAUtXqKus79t/FS",
+                new ArrayList<>(List.of(
+                        new Discovery("123", expectedTimestamp, "Pizza Place", "Sample Street 3", "https://google.com", "11:00 - 12:00", "04023457596", "https://google.com", "https://google.com", "Sample notes sample notes sample notes",
+                                new ArrayList<>(List.of("nature", "art"))),
+                        new Discovery("456", expectedTimestamp, "Sushi Place", "Sample Street 4", "https://google.com", "11:00 - 12:00", "04023457596", "https://google.com", "https://google.com", "Sample notes sample notes sample notes",
+                                new ArrayList<>(List.of("drink", "nature"))),
+                        new Discovery(
+                                "1234", expectedTimestamp, "Fun Discovery", "Fun address Str. 12, 20253 Hamburg",null, null, "345678976543", "https://google.com", null, "Fun notes fun notes fun notes fun notes", new ArrayList<>(
+                                List.of("restaurant","art"))
+
+        )))));
+
+        Discovery expectedDiscovery = new Discovery(
+                "1234", expectedTimestamp, "Fun Discovery", "Fun address Str. 12, 20253 Hamburg",null, null, "345678976543", "https://google.com", null, "Fun notes fun notes fun notes fun notes", new ArrayList<>(
+                List.of("restaurant","art"))
+        );
+
+        //THEN
+        assertThat(response.getStatusCode(), is(HttpStatus.OK));
+        assertThat(response.getBody(), is(expectedDiscovery));
```

Comment:
Maybe test if the list has the added discovery?

Reply:
That is tested in 165, as the user needs to have the added discovery in his list